src/postgres/getClientConfig.ts

```diff
@@ -34,7 +34,11 @@ export async function getClientConfig(treeItem: PostgresServerTreeItem, database
             };
         }
     } else {
-        clientConfig = { connectionString: connectionString.connectionString };
+        let connectionString = parsedCS.connectionString;
+        if (!parsedCS.databaseName) {
+            connectionString += databaseName;
```

Comment:
Unfortunately it's not this simple - we have to really think about edge cases with the attached scenario because users could enter any type of connection string. For example, users could have a query in the uri, meaning you'd be appending to the query instead of the pathname and it wouldn't work. Also they may or may not have a trailing slash already in the uri.

Check out Node.js's built-in "URL" class. You should be able to do something like this, which is more robust:
```
url = new URL(connectionString)
url.pathname = databaseName;
url.toString()
```

Also add a few unit tests, similar to what we did for mongo: https://github.com/microsoft/vscode-cosmosdb/blob/24843d11283ad5ed820db90fec47e84f0f73b840/test/mongoConnectionStrings.test.ts#L77